### ISSUE TITLE
Bluebird - inferring type

### ISSUE THREAD

**Issue by jonasi** [flow-typed/flow-typed#1827]
neither seem to complain as of flow 0.65

```
// @flow

import Promise from 'bluebird';
(Promise.resolve(3): Promise<string>);
(new Promise(resolve => resolve(3)): Promise<string>);
```

but it does fail as expected using the builtin Promise definition

```
  4: (Promise.resolve(3): Promise<string>);
      ^^^^^^^^^^^^^^^^^^ Promise. Has some incompatible type argument with
  4: (Promise.resolve(3): Promise<string>);
                          ^^^^^^^^^^^^^^^ Promise
  Type argument `R` is incompatible:
      4: (Promise.resolve(3): Promise<string>);
                          ^ number. This type is incompatible with
      4: (Promise.resolve(3): Promise<string>);
                                      ^^^^^^ string
```

**Comment by jonasi** [comment]
Two independent change seem to "fix" it:

- change type param of `Bluebird$Promise` to remove covariance
- Replace usage of `$Promisable` with it's definition (`Promise<T> | T`) in method definitions

**Comment by emhagman** [comment]
This is still happening. Is there any progress?

**Comment by jps327** [comment]
This should get fixed for flow v0.104.0+ with this PR: https://github.com/flow-typed/flow-typed/pull/3854
